mmv1/third_party/terraform/resources/resource_sql_database_instance.go.erb

```diff
@@ -1361,6 +1384,27 @@ func resourceSqlDatabaseInstanceUpdate(d *schema.ResourceData, meta interface{})
 	desiredSetting := d.Get("settings")
 	var op *sqladmin.Operation
 	var instance *sqladmin.DatabaseInstance
+	// Check if the maintenance version is being updated, because patching maintenance version is an atomic operation and can not be
+	// performed with other fields, we first patch maintenance version before updating the rest of the fields.
+	if v, ok := d.GetOk("maintenance_version"); ok {
+		instance = &sqladmin.DatabaseInstance{MaintenanceVersion: v.(string)}
+		err = retryTimeDuration(func() (rerr error) {
+			op, rerr = config.NewSqlAdminClient(userAgent).Instances.Patch(project, d.Get("name").(string), instance).Do()
```

Comment:
Just want to confirm if `Patch` is the right method in this case, as `Update` was used in the previous version.  Sorry, I couldn't find which verb should be the right one for updating this field based on the API doc

Reply:
Both update and patch should work in general. Patch is the right method for this particular case.